client/client.go

```diff
@@ -2405,9 +2406,8 @@ func (c *Client) updateAlloc(update *structs.Allocation) {
 
 	// Reconnect unknown allocations
 	if update.ClientStatus == structs.AllocClientStatusUnknown && update.AllocModifyIndex > ar.Alloc().AllocModifyIndex {
-		update.ClientStatus = ar.AllocState().ClientStatus
-		update.ClientDescription = ar.AllocState().ClientDescription
-		c.AllocStateUpdated(update)
```

Comment:
Take a close look at the `AllocState` implementation; `Reconnect` is not a valid refactoring here because we've removed the locks and memoization that `AllocState` gives us.

Reply:
Tennessee